### PR TITLE
Thermostat Section (WIP)

### PR DESCRIPTION
**Pull request by justinGilmer**
Meat of the Thermostat section complete. 

Summary table.

WIP ( pitfalls, and basic description of the chosen thermostats )

Not ready for merge.

Are there any glaring mistakes? Any thermostats that should be included?

### CODE REVIEW

**Comment by justinGilmer** [comment]
Rough draft of the thermostat section is complete. However, it is a bit lengthy. Would like to get feedback on what, if any, should be removed or condensed. If any glaring mistakes are noticed, please let me know.

**Comment by davidlmobley** [comment]
OK, I'm going through your edits, and have some comments in a couple categories...

Big picture:
- I like it!
- I don't think you can make it much shorter without losing detail we need
- Probably needs more explicit recommendations/warnings (see below)
- Probably needs more references

Things which need some attention:
- *Velocity rescaling*: One of the big drawbacks, it says, is that particles do not know their previous thermal history and this makes dynamical properties impossible to measure. It is not explained and not obvious why this is as stated; you probably want at least a reference to further reading. Also, people interested in equilibrium properties (free energies, etc.) don't care about this, so it may be that you want more stringent disclaimers for these people. I think it would be safe to EXPLICITLY give a recommendation at the end of this paragraph such as, "We recommend that velocity rescaling thermostats such as (whatever we're calling this one) never be used for production level simulations, except perhaps during an initial equilibration phase, because they do not sample from the correct ensemble and lead to incorrect estimates of various computed properties [refs]." (I wrote this thinking about just item 1 within 4.4.3, but it probably applies to items 1-3). Maybe also we should be more specific about what we mean by "velocity rescaling" since the subsequent "Hoover" section is also explained as "velocity rescaling".  
- Do you need to separately note the "massive collision" and "gradual buffeting" versions of this?
- Maybe you want to say something more positive about Langevin; right now there's not really a warning against it nor anything that positive, except that the statement "Careful consideration must be taken when choosing the friction damping parameter" could be interpreted as a warning. 
- I wonder if we should come down a little more strongly at the end of this thermostat discussion in terms of recommending certain types to be used/avoided -- e.g. a lot of these really shouldn't be used for most purposes and we should perhaps say that clearly and concisely.
- Up for me writing a comment to include on how "small system" issues apply to certain types of free energy calculations (e.g. if I decouple a ligand, the decoupled ligand can trigger problems with thermostat issues)
- Should we be referencing the "block of ice" problem?

Minor details: I should edit a few things in detail when we are ready:
- Right now it says in the motivation that, basically, thermostats are used to change the temperature during a simulation, which is not really what we mean
- I should just adjust some grammar/clarity issues

Basically, for this last category, I can do it in a separate PR once we merge.  :)

**Comment by JIMonroe** [comment]
So it sounds like the writing here is mainly done, but it just needs a few tweaks.

I'm going through and approving pull requests to get things moving again.  We can come back to any pending edits and comments later.

**Comment by avisekdas** [comment]
Thanks for getting back to this. We need to finish few more sections. But
we are getting there.

I will comment on some other discussions as well.

Thanks,
Avisek

**Comment by davidlmobley** [comment]
Thanks, @JIMonroe !
